### PR TITLE
Support creating a `JUnitTestCase` from a `DefaultTestSet` alone

### PR DESCRIPTION
**Pull request by nickrobinson251**
- in particular make `PerfStats` optional
- cc @mcmcgrath13

### CODE REVIEW

**Comment by nickrobinson251** [comment]
xref https://relationalai.atlassian.net/browse/RAI-12694

**Review by quinnj — APPROVED** [review]

File: src/junit_xml.jl [hunk 1]
```diff
@@ -28,7 +28,7 @@ end
 mutable struct JUnitTestCase  # TestItem run
     const name::String
     counts::JUnitCounts
-    stats::PerfStats
+    stats::Union{PerfStats, Nothing} # Additional stats not available from the testset
     error_message::Union{String,Nothing} # Additional message to include in `<error>`
     logs::Union{Vector{UInt8},Nothing} # Captured logs from a `@testitem`
```
Comment (mcmcgrath13) [inline src/junit_xml.jl:33]: do logs need to be a vector of `UInt8`? Or is a string okay here? The way my logs are captured they are of type `String`

**Review by mcmcgrath13 — COMMENTED** [review]
Thank you!

File: src/junit_xml.jl (hunk 1 above)
Comment (mcmcgrath13) [reply src/junit_xml.jl:33]: Actually, I take that back. I end up writing everything to a buffer when it gets formatted, so it is a `Vector{UInt8}`

**Pull request merged**